### ISSUE TITLE
Wav2Vec2 Fine-tuning Tutorial

### ISSUE THREAD

**Issue by agemagician** [NVIDIA/NeMo#2406]
Hi,

As you currently don't have a tutorial for training wav2vec2 #2342, I tried to create a new yaml file to train wav2vec2.
However, I didn't succeed and I need your guidance and help.

I have created a Colab example that replicate my experiment here:
https://colab.research.google.com/drive/1kJQA0wm7EVLGsDlZmhCv-2siXXJvTpUt?usp=sharing

This is the current error:
```
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
[NeMo I 2021-06-25 14:10:52 exp_manager:216] Experiments will be logged at /content/nemo_experiments/Wav2vec_Pretrain/2021-06-25_14-10-52
[NeMo I 2021-06-25 14:10:52 exp_manager:563] TensorboardLogger has been set up
Traceback (most recent call last):
  File "wav2vec_pretrain.py", line 97, in <module>
    main()  # noqa pylint: disable=no-value-for-parameter
  File "/usr/local/lib/python3.7/dist-packages/nemo/core/config/hydra_runner.py", line 103, in wrapper
    strict=None,
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/utils.py", line 347, in _run_hydra
    lambda: hydra.run(
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/utils.py", line 201, in run_and_report
    raise ex
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/utils.py", line 198, in run_and_report
    return func()
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/utils.py", line 350, in <lambda>
    overrides=args.overrides,
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/hydra.py", line 112, in run
    configure_logging=with_log_configuration,
  File "/usr/local/lib/python3.7/dist-packages/hydra/core/utils.py", line 127, in run_job
    ret.return_value = task_function(task_cfg)
  File "wav2vec_pretrain.py", line 92, in main
    wav2vec_encoder_model = Wav2VecEncoderModel(cfg=cfg.model, trainer=trainer)
  File "/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/models/wav2vec/wav2vec_model.py", line 54, in __init__
    self.global_rank = 0
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1178, in __setattr__
    object.__setattr__(self, name, value)
AttributeError: can't set attribute
```

Your feedback is highly appreciated.

**Comment by titu1994** [comment]
Wav2Vec is currently untested, and the code needs a few updates to be compatible with latest pytorch lightning. 
For now, you can comment out the lines here - 

https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/models/wav2vec/wav2vec_model.py#L54-L60


@SeanNaren fyi. If you'd like I can take a stab at updating it.

**Comment by agemagician** [comment]
I have commented this part of the code, but no there is another error:

```
omegaconf.errors.ConfigKeyError: Key 'labels' not in 'Wav2VecEncoderModelConfig'
	full_key: labels
	reference_type=Optional[Wav2VecEncoderModelConfig]
	object_type=Wav2VecEncoderModelConfig
```

I also tried to change the Yaml file and the code to use :
https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_to_text.py
But it doesn't work because there is no encoder module for wav2vec like "nemo.collections.asr.modules.ConvASREncoder"

**Comment by titu1994** [comment]
Wav2Vec does not utilize speech_to_text.py script, so that script is not compatible with it. 

It is currently not a supported model, therefore it's not tested well enough for general use. We will consider adding proper support for it in the future. @okuchaiev fyi

**Comment by okuchaiev** [comment]
well, this is something where we'd really welcome community contributions

**Comment by khursani8** [comment]
Hi, sorry for interrupting, I'm not sure if NeMo suppose to depend on external library for model creation, if ok can use https://pytorch.org/audio/stable/models.html#wav2vec2model or maybe follow their code style for adding pretrained models from huggingface and fairseq

**Comment by titu1994** [comment]
We'll look into it, but it's currently not a high priority model that we are focusing on.

**Comment by titu1994** [comment]
There's CTC finetuning tutorial which should be more or less sufficient for Wav2Vec finetuning once it's fixed in #3297
